### ISSUE TITLE
Duplicate complex  Field types in records fail to decode

### ISSUE THREAD

**Issue by tomijaga** [Gekctek/motoko_candid#7]
Hey @Gekctek,

I recently got an [issue ticket](https://github.com/NatLabs/serde/issues/9) from a package that uses your library to decode the candid types. Identified by the ticket's creator, the candid library fails to decode duplicate complex (non-primitive) types within a record.

Running this piece of code returns a `null` value:

```motoko
    type User = {
        name: Text;
        age: Nat;
    };

    type Record = {
        first: User;
        second: User; // duplicate of 'first' field `User` type
    };

    let user_james: User = {
        name = "James";
        age = 23;
    };

    let user_steven: User = {
        name = "Steven";
        age = 24;
    };

    let record = {
        first = user_james;
        second = user_steven
    };

   let record_blob = to_candid(record);
   let decoded = Decoder.decode(record_blob);

   assert decoded == null;

```

**Comment by Gekctek** [comment]
@tomijaga  Ok, I have fixed this for both encoding and decoding

Fixed in 1.0.2

**Comment by tomijaga** [comment]
Thanks for quickly resolving the issue!
